sdk/logs/src/main/java/io/opentelemetry/sdk/logs/EventToSpanEventBridge.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.logs;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A processor that records events (i.e. log records with an attribute {@code event.name}) to the
+ * current span, if it is valid and {@link Span#isRecording()} is true.
+ */
+public class EventToSpanEventBridge implements LogRecordProcessor {
+
+  private static final AttributeKey<String> EVENT_NAME = AttributeKey.stringKey("event.name");
+  private static final AttributeKey<String> RESULT =
+      AttributeKey.stringKey("event_span_event_bridge.result");
+  private static final Attributes NOT_EVENT = Attributes.of(RESULT, "skipped_not_event");
+  private static final Attributes SPAN_INVALID = Attributes.of(RESULT, "skipped_span_invalid");
+  private static final Attributes SPAN_NOT_RECORDING =
+      Attributes.of(RESULT, "skipped_span_not_recording");
+  private static final Attributes BRIDGED = Attributes.of(RESULT, "success");
+
+  private final LongCounter processedCounter;
+
+  private EventToSpanEventBridge(MeterProvider meterProvider) {
+    this.processedCounter =
+        meterProvider
+            .meterBuilder(EventToSpanEventBridge.class.getName())
+            .build()
+            .counterBuilder("event_span_event_bridge.processed")
+            .build();
+  }
+
+  public static EventToSpanEventBridge create(MeterProvider meterProvider) {
+    return new EventToSpanEventBridge(meterProvider);
+  }
+
+  public static EventToSpanEventBridge create() {
+    return new EventToSpanEventBridge(MeterProvider.noop());
+  }
+
+  @Override
+  public void onEmit(Context context, ReadWriteLogRecord logRecord) {
+    LogRecordData logRecordData = logRecord.toLogRecordData();
+    String eventName = logRecordData.getAttributes().get(EVENT_NAME);
+    if (eventName == null) {
+      processedCounter.add(1, NOT_EVENT);
+      return;
+    }
+    if (!logRecordData.getSpanContext().isValid()) {
+      processedCounter.add(1, SPAN_INVALID);
+      return;
+    }
+    Span currentSpan = Span.current();
+    if (!currentSpan.isRecording()) {
+      processedCounter.add(1, SPAN_NOT_RECORDING);
+      return;
+    }
+    currentSpan.addEvent(
```

Comment:
So, just speaking aloud. This works because the event api is involved directly (e.g. not intermediated through a queue or otherwise). So, it has access to whatever the current span is, which would match the trace and span id in the event, if someone added the span context.

In other words, we could if we wanted to, assert `logRecordData.getSpanContext()` is the same as the current span context.

Reply:
Yes exactly. `LogRecordProcessors` are called synchronously from the point where the log was emitted in the application. This means its important for them to be fast and non-blocking, but grants access to the current context, including the active span.